# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/pkg_patch/pkg_grub.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# Copyright (C) 1998 - 2004 ROCK Linux Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- grub-0.93/lib/device.c.orig	Thu Sep  4 16:25:24 2003
+++ grub-0.93/lib/device.c	Thu Sep  4 16:23:46 2003
@@ -699,7 +699,7 @@
       return 0;
     }
   
-#if defined(__linux__) && (!defined(__GLIBC__) || \
+#if defined(__linux__) && (!defined(__dietlibc__)) && (!defined(__GLIBC__) || \
         ((__GLIBC__ < 2) || ((__GLIBC__ == 2) && (__GLIBC_MINOR__ < 1))))
   /* Maybe libc doesn't have large file support.  */
   {
--- grub-0.93/grub/asmstub.c~	Mon Dec  2 23:20:45 2002
+++ grub-0.93/grub/asmstub.c	Thu Sep  4 16:30:09 2003
@@ -934,7 +934,7 @@
     return BIOSDISK_ERROR_GEOMETRY;
 
   /* Seek to the specified location. */
-#if defined(__linux__) && (!defined(__GLIBC__) || \
+#if defined(__linux__) && (!defined(__dietlibc__)) && (!defined(__GLIBC__) || \
 	((__GLIBC__ < 2) || ((__GLIBC__ == 2) && (__GLIBC_MINOR__ < 1))))
   /* Maybe libc doesn't have large file support.  */
   {
